analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

```diff
@@ -0,0 +1,534 @@
+# Jo Lynne found the following status for 2/3 in the data warehouse:
+# C4106355-Tumor = Recurrence
+# C4113612-Tumor = Initial CNS Tumor
+# 7316UP-3705 = Not Reported
+# manually annotate them here and these will be updated in the v13 - then discard that part of the code
+
+# I also added the cancer_group into the match_id to use this later in the script
+# Also added the experimental_strategy to create unique events tagged to each assay 
+# so we won't miss samples with paired assays across multiple time points
+# if you remove this, print(length(unique(pbta$match_id))) will result to 2258, instead of 4293 thatis now
+# hope this makes and it is correct rational...
+
+pbta <- v12 %>%
+  filter(cohort == "PBTA",
+         !is.na(pathology_diagnosis),
+         !composition %in% c("Derived Cell Line", "PDX"),
+         !experimental_strategy %in% c("Methylation", "Targeted Sequencing")) %>%
+           # !is.na(cancer_group) # Discard NAs in cancer type
+  mutate(match_id = paste(cancer_group, sample_id, composition, experimental_strategy, tumor_descriptor, sep = "_")) %>% 
+  mutate(tumor_descriptor_final = case_when(grepl("C4106355-Tumor", sample_id) ~ "Recurrence", # to be updated in v13 as "Recurrence"
+                                            grepl("C4113612-Tumor", sample_id) ~ "Diagnosis", # to be updated in v13 as "Initial CNS Tumor"
+                                            grepl("Primary Tumor", tumor_descriptor) ~ "Diagnosis",
+                                            grepl("Initial CNS Tumor", tumor_descriptor) ~ "Diagnosis",
+                                            grepl('Progressive Disease Post-Mortem', tumor_descriptor) ~ 'Deceased', 
+                                            TRUE ~ tumor_descriptor))
+
+print(pbta %>% count(tumor_descriptor_final))
+
+# Save pbta after filter 
+write_tsv(pbta, path = output_cohort_pbta)
+```
+
+
+## Filter based on both genomic and transcriptomic assays
+Let's select all patient samples containing both genomic and transcriptomic assays.
+
+```{r assays, echo=TRUE}
+
+print(length(unique(pbta$Kids_First_Participant_ID)))
+print(length(unique(pbta$match_id)))
+
+assays <- pbta %>%
+  select(Kids_First_Participant_ID, tumor_descriptor_final, match_id, experimental_strategy, cancer_group) %>%
+  unique() %>%
+  arrange(Kids_First_Participant_ID)
+
+
+# we perform join here using the Kids_First_Participant_ID and match_id
+# THIS PART I AM NOT SURE OF 
+paired_assays <- assays %>%
+  group_by(Kids_First_Participant_ID, match_id) %>%
+  summarise(experimental_strategy = str_c(experimental_strategy, collapse = ";")) %>%
+  mutate(experimental_strategy_sum = case_when(grepl("WGS", experimental_strategy) ~ "DNA",
+                                                grepl("WXS", experimental_strategy) ~ "DNA", 
+                                                grepl("RNA-Seq", experimental_strategy) ~ "RNA",
+                                                TRUE ~ experimental_strategy)) 
```

Comment:
Is `experimental_strategy_sum` supposed to be indicating whether a sample has RNA, DNA, or both? If so, this logic should work: 
```suggestion
  mutate(experimental_strategy_sum = case_when(grepl("RNA-Seq", experimental_strategy) & grepl("WGS|WXS", experimental_strategy) ~ "DNA,RNA",
                                                grepl("WXS|WGS", experimental_strategy) ~ "DNA", 
                                                grepl("RNA-Seq", experimental_strategy) ~ "RNA"
                                                )) 
```